cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -74,9 +87,29 @@ public void initialize(WorkflowContext context) throws Exception {
              programType != null ? programType.name() : null, programName);
   }
 
+  public void setTxClient(TransactionSystemClient txClient) {
```

Comment:
It think we don't need TransactionSystemClient in this class at all. As I mentioned above, the workflow driver should be the one controlling the transaction.